jose.exceptions.JWTError: Error decoding token claims

[2025-09-09 07:42:25 +0200] [7] [INFO] Starting gunicorn 23.0.0
[2025-09-09 07:42:25 +0200] [7] [DEBUG] Arbiter booted
[2025-09-09 07:42:25 +0200] [7] [INFO] Listening at: http://0.0.0.0:8000 (7)
[2025-09-09 07:42:25 +0200] [7] [INFO] Using worker: uvicorn.workers.UvicornWorker
[2025-09-09 07:42:25 +0200] [8] [INFO] Booting worker with pid: 8
[2025-09-09 07:42:25 +0200] [9] [INFO] Booting worker with pid: 9
[2025-09-09 07:42:25 +0200] [7] [DEBUG] 2 workers
[2025-09-09 07:42:28 +0200] [9] [INFO] Started server process [9]
[2025-09-09 07:42:28 +0200] [9] [INFO] Waiting for application startup.
[2025-09-09 07:42:28 +0200] [8] [INFO] Started server process [8]
[2025-09-09 07:42:28 +0200] [8] [INFO] Waiting for application startup.
[2025-09-09 07:42:28 +0200] [9] [INFO] Application startup complete.
[2025-09-09 07:42:28 +0200] [8] [INFO] Application startup complete.
[2025-09-09 07:43:20 +0200] [8] [ERROR] Exception in ASGI application
Traceback (most recent call last):
  File "/venv/lib/python3.11/site-packages/jose/jws.py", line 180, in _load
    signing_input, crypto_segment = jwt.rsplit(b".", 1)
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ValueError: not enough values to unpack (expected 2, got 1)

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/venv/lib/python3.11/site-packages/jose/jwt.py", line 238, in get_unverified_claims
    claims = jws.get_unverified_claims(token)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/jose/jws.py", line 128, in get_unverified_claims
    header, claims, signing_input, signature = _load(token)
                                               ^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/jose/jws.py", line 184, in _load
    raise JWSError("Not enough segments")
jose.exceptions.JWSError: Not enough segments

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/venv/lib/python3.11/site-packages/uvicorn/protocols/http/h11_impl.py", line 429, in run_asgi
    result = await app(  # type: ignore[func-returns-value]
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/uvicorn/middleware/proxy_headers.py", line 78, in __call__
    return await self.app(scope, receive, send)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/fastapi/applications.py", line 1054, in __call__
    await super().__call__(scope, receive, send)
  File "/venv/lib/python3.11/site-packages/starlette/applications.py", line 112, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/venv/lib/python3.11/site-packages/starlette/middleware/errors.py", line 187, in __call__
    raise exc
  File "/venv/lib/python3.11/site-packages/starlette/middleware/errors.py", line 165, in __call__
    await self.app(scope, receive, _send)
  File "/venv/lib/python3.11/site-packages/starlette/middleware/exceptions.py", line 62, in __call__
    await wrap_app_handling_exceptions(self.app, conn)(scope, receive, send)
  File "/venv/lib/python3.11/site-packages/starlette/_exception_handler.py", line 53, in wrapped_app
    raise exc
  File "/venv/lib/python3.11/site-packages/starlette/_exception_handler.py", line 42, in wrapped_app
    await app(scope, receive, sender)
  File "/venv/lib/python3.11/site-packages/starlette/routing.py", line 714, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/venv/lib/python3.11/site-packages/starlette/routing.py", line 734, in app
    await route.handle(scope, receive, send)
  File "/venv/lib/python3.11/site-packages/starlette/routing.py", line 288, in handle
    await self.app(scope, receive, send)
  File "/venv/lib/python3.11/site-packages/starlette/routing.py", line 76, in app
    await wrap_app_handling_exceptions(app, request)(scope, receive, send)
  File "/venv/lib/python3.11/site-packages/starlette/_exception_handler.py", line 53, in wrapped_app
    raise exc
  File "/venv/lib/python3.11/site-packages/starlette/_exception_handler.py", line 42, in wrapped_app
    await app(scope, receive, sender)
  File "/venv/lib/python3.11/site-packages/starlette/routing.py", line 73, in app
    response = await f(request)
               ^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/fastapi/routing.py", line 291, in app
    solved_result = await solve_dependencies(
                    ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/fastapi/dependencies/utils.py", line 638, in solve_dependencies
    solved = await call(**solved_result.values)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/worker/app/src/chanjo2/auth.py", line 56, in get_token
    claims = jwt.get_unverified_claims(token)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/venv/lib/python3.11/site-packages/jose/jwt.py", line 240, in get_unverified_claims
    raise JWTError("Error decoding token claims.")
jose.exceptions.JWTError: Error decoding token claims.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -1,7 +1,9 @@
 ## [unreleased]
 ### Changed
-- Validate and persist OIDC token in cookie for report and overview endpoints (#492)
 - Condensed summary endpoint not protected by auth for now (#492)
+### Fixed
+- Validate and persist OIDC token in cookie for report and overview endpoints (#492)
+- Fix app crashing when try validating token claims in absence of token (#494)
 
 ## [3.8]
 ### Changed 

--- a/src/chanjo2/auth.py
+++ b/src/chanjo2/auth.py
@@ -52,11 +52,11 @@ async def get_token(request: Request) -> Tuple[str, datetime.datetime]:
     if not token:
         raise HTTPException(status_code=401, detail="Missing id_token")
 
-    # Optional: log unverified claims
-    claims = jwt.get_unverified_claims(token)
-    print(f"Received a token with audience: {claims.get('aud')}")
-
     try:
+        # Optional: log unverified claims
+        claims = jwt.get_unverified_claims(token)
+        print(f"Received a token with audience: {claims.get('aud')}")
+
         # Fetch JWKS keys
         async with httpx.AsyncClient() as client:
             resp = await client.get(JWKS_URL)

